pmlc/dialect/pxa/transforms/autotile.h

```diff
@@ -70,17 +70,25 @@ class UnionGenerator<Single> {
   Single single;
 };
 
+// Empty target info. Subclasses for different targets.
+class TargetInfo {
+};
+
 // A tile cost model is a functor from an array of tile sizes (i.e.
 // ArrayRef<int64_t>) to a double, which is 'inf' for infeasible tilings.  For
 // example:
 
-inline double DummyCostModel(ArrayRef<int64_t> tile) { return 1.0; }
+inline double DummyCostModel(mlir::AffineParallelOp op, ArrayRef<int64_t> tile, const TargetInfo& info) {
+  return 1.0;
+}
 
 // Given a generator and cost model, find the best tile size, return empty
 // tiling when all tiles are infeasible
 template <typename Generator, typename CostModel>
-llvm::SmallVector<int64_t, 8> findBestTileSize(const Generator &generator,
+llvm::SmallVector<int64_t, 8> findBestTileSize(mlir::AffineParallelOp op,
+                                               const Generator &generator,
                                                const CostModel &costModel,
+                                               const TargetInfo &targetInfo,
```

Comment:
I don't think we need to pass anything extra here, the `CostModel` callback can contain any target-specific context you need.

Imagine this:

```
struct MyCostModel {
  MyCostModel(unsigned numCores) : numCores(numCores) {}

  double operator()(ArrayRef<int64_t> tile) {
    // use numCores to compute cost
    return numCores * 3;
  }

  unsigned numCores;
};
```

Then to use it:
```
auto tileSize = findBestTileSize(MyGenerator{}, MyCostModel{10}, *ranges);
```